ext/url/lib.rs

```diff
@@ -21,231 +30,448 @@ deno_core::extension!(
+  fn host(&self, #[webidl] value: String) {
+    quirks::set_host(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[string]
+  fn hostname(&self) -> String {
+    quirks::hostname(&self.0.borrow()).to_string()
+  }
+
+  #[setter]
+  fn hostname(&self, #[webidl] value: String) {
+    quirks::set_hostname(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[string]
+  fn href(&self) -> String {
+    quirks::href(&self.0.borrow()).to_string()
+  }
+
+  #[setter]
+  fn href(&self, #[webidl] value: String) {
+    quirks::set_href(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[string]
+  fn origin(&self) -> String {
+    quirks::origin(&self.0.borrow())
+  }
+
+  #[getter]
+  #[string]
+  fn password(&self) -> String {
+    quirks::password(&self.0.borrow()).to_string()
+  }
+
+  #[setter]
+  fn password(&self, #[webidl] value: String) {
+    quirks::set_password(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[string]
+  fn pathname(&self) -> String {
+    quirks::pathname(&self.0.borrow()).to_string()
+  }
+
+  #[setter]
+  fn pathname(&self, #[webidl] value: String) {
+    quirks::set_pathname(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[string]
+  fn port(&self) -> String {
+    quirks::port(&self.0.borrow()).to_string()
+  }
+
+  #[setter]
+  fn port(&self, #[webidl] value: String) {
+    quirks::set_port(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[string]
+  fn protocol(&self) -> String {
+    quirks::protocol(&self.0.borrow()).to_string()
+  }
+
+  #[setter]
+  fn protocol(&self, #[webidl] value: String) {
+    quirks::set_protocol(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[string]
+  fn search(&self) -> String {
+    quirks::search(&self.0.borrow()).to_string()
+  }
+
+  #[setter]
+  fn search(&self, #[webidl] value: String) {
+    quirks::set_search(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[string]
+  fn username(&self) -> String {
+    quirks::username(&self.0.borrow()).to_string()
+  }
+
+  #[setter]
+  fn username(&self, #[webidl] value: String) {
+    quirks::set_username(&mut self.0.borrow_mut(), &value);
+  }
+
+  #[getter]
+  #[global]
+  fn searchParams(&self, scope: &mut v8::HandleScope) -> v8::Global<v8::Object> {
+    self.1.get(scope, || {
+      let repr =
+        form_urlencoded::parse(quirks::search(&self.0.borrow()).as_bytes())
+          .into_owned()
+          .collect();
+      URLSearchParams {
+        inner_url: Some(self.0.clone()),
+        repr: RefCell::new(repr),
       }
-    }
-    Err(_) => ParseStatus::Err as u32,
+    })
   }
-}
 
-#[derive(Eq, PartialEq, Debug)]
-#[repr(u8)]
-pub enum UrlSetter {
-  Hash = 0,
-  Host = 1,
-  Hostname = 2,
-  Password = 3,
-  Pathname = 4,
-  Port = 5,
-  Protocol = 6,
-  Search = 7,
-  Username = 8,
+  #[string]
+  fn toString(&self) -> String {
+    self.0.borrow().to_string()
+  }
+
+  #[string]
+  fn toJSON(&self) -> String {
+    self.0.borrow().to_string()
+  }
```

Comment:
```suggestion
  #[string]
  fn to_string(&self) -> String {
    self.0.borrow().to_string()
  }

  #[rename("toJSON")]
  #[string]
  fn to_json(&self) -> String {
    self.0.borrow().to_string()
  }
```

Object wrap does automatic name conversion to camelCase